virtool/analyses/data.py

```diff
@@ -307,16 +310,15 @@ async def blast(self, analysis_id: str, sequence_index: int) -> Optional[str]:
             {"_id": analysis_id}, ["ready", "workflow", "results", "sample"]
         )
 
-        if document["workflow"] != "nuvs":
-            raise ResourceConflictError("Not a NuVs analysis")
+        aws = []
 
-        if not document["ready"]:
-            raise ResourceConflictError("Analysis is still running")
+        aws.append(check_analysis_workflow(document["workflow"]))
 
-        sequence = find_nuvs_sequence_by_index(document, sequence_index)
+        aws.append(check_if_analysis_running(document["ready"]))
 
-        if sequence is None:
-            raise ResourceNotFoundError()
+        aws.append(check_analysis_nuvs_sequence(document, sequence_index))
+
+        await wait_for_checks(*aws)
```

Comment:
No need for list or multiple expressions here. You can do this in a single call to `wait_for_checks`. Read up on `*` and variadic arguments in Python.